src/events/etcd.py

```diff
@@ -124,13 +124,24 @@ def _on_start(self, event: ops.StartEvent) -> None:
                     self.charm.state.cluster.update({"authentication": "enabled"})
                 except (EtcdAuthNotEnabledError, EtcdUserManagementError) as e:
                     logger.error(e)
-                    self.charm.set_status(Status.AUTHENTICATION_NOT_ENABLED)
-                    return
+                    raise
         elif (
             self.charm.state.unit_server.member_endpoint
             in self.charm.state.cluster.cluster_members
         ):
             # this unit has been added to the etcd cluster
+            if not self.charm.state.cluster.auth_enabled:
+                if self.charm.unit.is_leader():
+                    # if enabling auth failed on first cluster startup, we want to retry now
+                    try:
+                        self.charm.cluster_manager.enable_authentication()
```

Comment:
Could you add a 3-attempts retry with a 1sec backoff prior to definitively failing?